src/modules/esl-share/core/esl-share-button.ts

```diff
@@ -51,31 +104,59 @@ export class ESLShareButton extends ESLBaseElement {
   protected override attributeChangedCallback(attrName: string, oldVal: string, newVal: string): void {
     if (!this.connected || oldVal === newVal) return;
     if (attrName === 'action') this.updateAction();
+    if (attrName === 'name') this.updateName();
   }
 
   protected override connectedCallback(): void {
     super.connectedCallback();
+    this.init();
+  }
+
+  /** Initializes the button */
+  protected init(): void {
+    if (this.ready) return;
+
     this.initA11y();
     this.updateAction();
+    this.toggleAttribute('ready', true);
   }
 
   /** Sets initial a11y attributes */
   protected initA11y(): void {
     if (!this.hasAttribute('role')) this.setAttribute('role', 'button');
-    if (this.getAttribute('role') === 'button' && !this.hasAttribute('tabindex')) {
+    if (!this.hasAttribute('tabindex') && this.getAttribute('role') === 'button') {
       this.tabIndex = 0;
     }
   }
 
+  /** Initializes the button content */
+  protected initContent(): void {
```

Comment:
Cosmetic: for future updates
Rewrite content by attribute marker

```html
<!-- This one will not change -->
<esl-share-button name="facebook"> Facebook </esl-share-button>

<!-- Here the contact will be replaced -->
<esl-share-button name="facebook" load-content> Facebook </esl-share-button>
```

Or as an alternative:
```html
<esl-share-button name="facebook"><span esl-share-icon><!--icon dynamically appears here--></span> Facebook </esl-share-button>
```